File: Python/src/my_deque.py
class MyDeque:
    def __init__(self):
        self.deque = []

    def add_first(self, value):
        # Adds an element to the front of the deque
        self.deque.insert(0, value)

    def add_last(self, value):
        # Adds an element to the end of the deque
        self.deque.append(value)

    def remove_first(self):
        # Removes and returns the first element from the deque
        if not self.deque:
            raise IndexError("Deque is empty")
        return self.deque.pop(0)

    def remove_last(self):
        # Removes and returns the last element from the deque
        if not self.deque:
            raise IndexError("Deque is empty")
        return self.deque.pop()

    def size(self):
        # Returns the size of the deque
        return len(self.deque)

File: CSharp/src/MyDeque.cs
namespace Datastructure;

using System;

public class MyDeque<T>
{
    private LinkedList<T> _deque;

    public MyDeque()
    {
        _deque = new LinkedList<T>();
    }

    // Add an element to the front of the deque
    public void AddFirst(T item)
    {
        _deque.AddFirst(item);
    }

    // Add an element to the end of the deque
    public void AddLast(T item)
    {
        _deque.AddLast(item);
    }

    // Remove and return the element at the front
    public T RemoveFirst()
    {
        if (_deque.Count == 0)
        {
            throw new InvalidOperationException("Deque is empty.");
        }

        var value = _deque.First.Value;
        _deque.RemoveFirst();
        return value;
    }

    // Remove and return the element at the end
    public T RemoveLast()
    {
        if (_deque.Count == 0)
        {
            throw new InvalidOperationException("Deque is empty.");
        }

        var value = _deque.Last.Value;
        _deque.RemoveLast();
        return value;
    }

    // Peek at the first element without removing it
    public T PeekFirst()
    {
        if (_deque.Count == 0)
        {
            throw new InvalidOperationException("Deque is empty.");
        }

        return _deque.First.Value;
    }

    // Peek at the last element without removing it
    public T PeekLast()
    {
        if (_deque.Count == 0)
        {
            throw new InvalidOperationException("Deque is empty.");
        }

        return _deque.Last.Value;
    }

    // Check if the deque is empty
    public bool IsEmpty()
    {
        return _deque.Count == 0;
    }

    // Return the number of elements in the deque
    public int Count()
    {
        return _deque.Count;
    }
}
